autoload/jedi.vim

```diff
@@ -569,10 +569,17 @@ function! jedi#complete_string(autocomplete) abort
         set completeopt+=menuone
         set completeopt-=menu
         if &completeopt !~# 'noinsert\|noselect'
-            if g:jedi#popup_select_first
-                set completeopt+=noinsert
+            " Patch 775 introduced noinsert and noselect, previously these
+            " options didn't exist. Setting them in earlier versions results in
+            " errors (E474).
+            if has("patch-7.4-775")
+                if g:jedi#popup_select_first
+                    set completeopt+=noinsert
+                else
+                    set completeopt+=noselect
+                endif
             else
-                set completeopt+=noselect
+                set completeopt+=longest
```

Comment:
Not sure about using longest here then instead.. but I currently do not remember what was used before etc.
Could be added to the comment.

Reply:
I just added longest, because the tests pass on travis with it. I'm pretty sure we used it as default back in the days... However the whole `popup_select_first` stuff is older than `noselect` and `noinsert` and I'm therefore not even sure if that stuff is needed anymore, because these two settings were probably exactly created, because of the problems I had with VIM's autocompletion. Do you still think I should add that as a comment?

Reply:
I added a comment. Let me know if there are other things that are unclear/that you would like to see changed.